src/hotspot/cpu/riscv/vmreg_riscv.inline.hpp

```diff
@@ -26,20 +26,17 @@
 #ifndef CPU_RISCV_VM_VMREG_RISCV_INLINE_HPP
 #define CPU_RISCV_VM_VMREG_RISCV_INLINE_HPP
 
-inline VMReg RegisterImpl::as_VMReg() const {
-  if (this == noreg) {
-    return VMRegImpl::Bad();
-  }
-  return VMRegImpl::as_VMReg(encoding() * RegisterImpl::max_slots_per_register);
+inline VMReg Register::RegisterImpl::as_VMReg() const {
+  return VMRegImpl::as_VMReg(encoding() * Register::max_slots_per_register);
 }
 
-inline VMReg FloatRegisterImpl::as_VMReg() const {
-  return VMRegImpl::as_VMReg((encoding() * FloatRegisterImpl::max_slots_per_register) +
+inline VMReg FloatRegister::FloatRegister::FloatRegisterImpl::as_VMReg() const {
```

Comment:
Shouldn't this be: inline VMReg FloatRegister::FloatRegisterImpl::as_VMReg() const?

Reply:
Yes, fixed.